neuromation/cli/utils.py

```diff
@@ -489,6 +489,25 @@ def parse_resource_for_sharing(uri: str, root: Root) -> URL:
     return uri_from_cli(uri, root.username, allowed_schemes=("storage", "image", "job"))
 
 
+async def parse_and_resolve_resource_for_sharing(uri: str, root: Root) -> URL:
+    """ Parses the neuromation resource URI string + resolves job-name to job-ID.
+    Available schemes: storage, image, job. For image URIs, tags are not allowed.
+    """
+    uri_obj = parse_resource_for_sharing(uri, root)
```

Comment:
`parse_resource_for_sharing()` is only used here. I suggest to inline it.

Reply:
let's keep it as-is for now: since it's non-async function that encapsulates quite large logic of uri parsing, which is already covered with unit tests, I decided to simply wrap it with an async function `parse_and_resolve_resource_for_sharing` rather than replace it with another function. Let's keep this refactoring for the future (if needed) in order for this PR to remain smaller.

Reply:
Okay.